Reflection_Engine/Compute/RunExtentionMethod.cs

```diff
@@ -46,13 +46,12 @@ public static object RunExtentionMethod(object target, string methodName)
                 return m_PreviousInvokedMethods[key].Invoke(null, new object[] { target });
 
             // Otherwise, search for the method and call it if found
-            foreach (MethodInfo method in type.ExtentionMethods(methodName))
```

Comment:
So, let me check I I get what is happening. Since we removed the loop, why are we sorting first?
Can't we just take the correct overload?

Reply:
so, this line contains quite a bit of information, maybe a bit to much, but to break it down for clarity:

- `ExtentionMethods(type, methodName)` Returns all methods with the given name that has the first argument of a type matching (being assignable from) the provided type
- `Where(x => x.GetParameters().Length == 1)` filters out so we only take the methods that have exactly one argument, as we know by the context that that is the only methods we want for this.
- `SortExtensionMethods(type)` sorts the methods based on "closeness" to the provided type. (assume this is the one you are questioning the most). This is currently quite simplistic for in that it ranks exact type match high and non-exact type match low. We will want to expand upon this going forward.
- `FirstOrDefault()` then simply returns the best match

As an example of what the sorting is there for:

Say we have a method `Foo(IFramingElement)` and a method `Foo(Beam)`. 
If one calls `RunExtensionMethod` providing a `Column` it will only match and call the first method,  `Foo(IFramingElement)`. 
But, if one calls `RunExtensionMethod` providing a `Beam` it will match with both methods, and this is where we need the sorting, to make sure we rank the method `Foo(Beam)` higher than `Foo(IFramingElement)`. What we are trying to do is to match the method overloading that c# does automatically for you normally.. 
Both methods would be "a correct" overload, but we need to figure out which of them that is most correct, if that makes sense?

Reply:
As per our last chat, I see the picture - I'll get this approved!